step-plans/step-plans-yaml-parser/src/main/java/step/plans/parser/yaml/editor/LineNumberByJsonPointerResolver.java

```diff
@@ -0,0 +1,322 @@
+/*******************************************************************************
+ * Copyright (C) 2020, exense GmbH
+ *
+ * This file is part of STEP
+ *
+ * STEP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * STEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with STEP.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package step.plans.parser.yaml.editor;
+
+import com.fasterxml.jackson.core.JsonLocation;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.*;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLParser;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+import com.jayway.jsonpath.spi.json.JacksonJsonNodeJsonProvider;
+import com.jayway.jsonpath.spi.mapper.JacksonMappingProvider;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class LineNumberByJsonPointerResolver {
+
+    private static final Pattern specialChars = Pattern.compile("[\\s~!@#$%^&*()_+\\-=\\[\\]{};':\"\\\\|,.<>/?]+");
+
+    public LineNumberByJsonPointerResolver() {
+    }
+
+    public List<JsonPointerSourceLine> findLineNumbers(List<String> jsonPointers, String source) {
+        List<JsonPointerSourceLine> result = new ArrayList<>();
+
+        // TODO: maybe optimize and avoid creation of new ObjectMapper and factory for each source
```

Comment:
@iegorov777 yes, I don't think the creation of com.jayway.jsonpath.Configuration has to be done for each invocation of findLineNumbers. If it's thread safe a single instance should be enough